frontend/src/stimulus/controllers/dynamic/reporting/page.controller.ts

```diff
@@ -0,0 +1,824 @@
+    return this.groupBys.group_by_container_ids().filter((container:string) =>
+      jQuery(`#${container}`).find('[data-group-by]').length > 0);
+  }
+
+  private initializeControls() {
+    this.controls = {
+      attach_settings_callback: (element:JQuery, callback:(response:string) => void) => {
+        this.attachSettingsCallback(element, callback);
+      },
+
+      clear_query: (e:Event) => {
+        e.preventDefault();
+        this.filters.clear();
+        this.groupBys.clear();
+      },
+
+      observe_click: (elementId:string, callback:(e:Event) => void) => {
+        jQuery(`#${elementId}`).on('click', callback);
+      },
+
+      update_result_table: (response:string) => {
+        jQuery('#result-table').html(response);
+        this.initTableSorter();
+      },
+
+      toggle_delete_form: (e:Event) => {
+        e.preventDefault();
+        const offset = jQuery('#query-icon-delete').offset()?.left || 0;
+        jQuery('#delete_form').css('left', `${offset}px`).toggle();
+      },
+
+      toggle_save_as_form: (e:Event) => {
+        e.preventDefault();
+        const offset = jQuery('#query-icon-save-as').offset()?.left || 0;
+        jQuery('#save_as_form').css('left', `${offset}px`).toggle();
+      },
+    };
+
+    // Bind control events
+    if (jQuery('#query_saved_name').length) {
+      if (jQuery('#query_saved_name').attr('data-is_new')) {
+        if (jQuery('#query-icon-delete').length) {
+          this.controls.observe_click('query-icon-delete', this.controls.toggle_delete_form);
+          this.controls.observe_click('query-icon-delete-cancel', this.controls.toggle_delete_form);
+          jQuery('#delete_form').hide();
+        }
+
+        if (jQuery('#query-breadcrumb-save').length) {
+          this.controls.attach_settings_callback(jQuery('#query-breadcrumb-save'), this.controls.update_result_table);
+        }
+      }
+    }
+
+    this.controls.observe_click('query-icon-save-as', this.controls.toggle_save_as_form);
+    this.controls.observe_click('query-icon-save-as-cancel', this.controls.toggle_save_as_form);
+
+    jQuery('#save_as_form').hide();
+
+    this.controls.attach_settings_callback(jQuery('#query-icon-save-button'), (newLocation:string) => {
+      document.location.href = newLocation;
+    });
+
+    this.controls.attach_settings_callback(jQuery('#query-icon-apply-button'), this.controls.update_result_table);
+
+    this.controls.observe_click('query-link-clear', this.controls.clear_query);
+  }
+
+  private attachSettingsCallback(element:JQuery, callback:(response:string) => void) {
+    if (!element || !element.length) {
+      return;
+    }
+
+    const failureCallback = (response:JQuery.jqXHR) => {
+      jQuery('#result-table').html('');
+      this.defaultFailureCallback(response);
+    };
+
+    element.on('click', (e:Event) => {
+      e.preventDefault();
+      this.sendSettingsData(jQuery(e.target as HTMLElement).attr('data-target') || '', callback, failureCallback);
```

Comment:
It isn't safe if a <span> or icon inside a button is clicked.
My suggestion would be:
`jQuery(e.target as HTMLElement).closest('[data-target]').attr('data-target')`